src/query/functions/src/scalars/timestamp/src/datetime.rs

```diff
@@ -1451,6 +1451,21 @@ fn register_to_number_functions(registry: &mut FunctionRegistry) {
             }
         }),
     );
+
+    registry.register_passthrough_nullable_1_arg::<DateType, UInt16Type, _, _>(
+        "to_iso_year",
+        |_, _| FunctionDomain::Full,
+        vectorize_with_builder_1_arg::<DateType, UInt16Type>(|val, output, ctx| {
+            match ToNumberImpl::eval_date::<ToISOYear, _>(val, ctx.func_ctx.tz.clone()) {
+                Ok(t) => output.push(t),
+                Err(e) => {
+                    ctx.set_error(output.len(), format!("cannot parse to type `Date`. {}", e));
```

Comment:
Parse to type Date? Maybe assign func name is better

Reply:
This is in function `eval_date `, we will parse the val into date and get the year value.